rootfs/ingest/python/qserv/queue.py

```diff
@@ -169,77 +185,91 @@ def lock_contributions(self) -> list:
+
+    def release_locked_contributions(self, ingest_success: bool) -> None:
+        """Mark contributions as "succeed" in contribution queue if super-transaction
             has been successfully commited
             Release contributions in queue when the super-transaction has been aborted
 
             WARN: this operation will be retried until it succeed
             so that contribution queue state is consistent with ingest state
-        Returns
-        -------
-        Integer number, String
         """
         if ingest_success:
-            query = self.queue.update().values(succeed=1)
+            query = self.queue_table.update().values(succeed=1)
             logging.debug("Mark contributions as 'succeed' in queue")
-            query = self.queue.update().values(succeed=1)
+            query = self.queue_table.update().values(succeed=1)
         else:
             logging.debug("Unlock contributions in queue")
-            query = self.queue.update().values(locking_pod=None)
+            query = self.queue_table.update().values(locking_pod=None)
 
-        query = query.where(self.queue.c.locking_pod == self.pod)
-        query = query.where(self.queue.c.succeed.is_(None))
+        query = query.where(self.queue_table.c.locking_pod == self.pod)
+        query = query.where(self.queue_table.c.succeed.is_(None))
 
         wait_sec = 1
         while True:
             try:
                 _LOG.debug("Query: %s", query)
                 self.engine.execute(query)
                 break
-            except:
+            except Exception:
                 _LOG.critical("Retry failed query %s", query)
                 time.sleep(wait_sec)
                 # Sleep for longer and longer
                 wait_sec = increase_wait_time(wait_sec)
 
-    def _is_queue_empty(self):
-        return (self.current_table == None)
-
-    def get_noningested_contributions(self):
-        """Return all contribution in queue not successfully loaded for current database
-        """
-        query = select([self.queue.c.database,
-                        self.queue.c.chunk_id,
-                        self.queue.c.chunk_file_path,
-                        self.queue.c.is_overlap,
-                        self.queue.c.table])
-        query = query.where(self.queue.c.succeed.is_(None))
-        query = query.where(self.queue.c.database == self.contribution_metadata.database)
+    def _is_queue_empty(self) -> bool:
+        if self.current_table is None:
+            return True
+        else:
+            return False
+
+    def select_noningested_contributions(self):
+        """Return all contribution in queue not successfully loaded for current database"""
+        query = select(
+            [
+                self.queue_table.c.database,
+                self.queue_table.c.chunk_id,
+                self.queue_table.c.filepath,
+                self.queue_table.c.is_overlap,
+                self.queue_table.c.table,
+            ]
+        )
```

Comment:
sqlalchemy 2 will stop accepting a list in `select`, may be better to switch to all positional arguments.